src/HDF5.jl

```diff
@@ -2572,6 +2555,18 @@ const ASCII_ATTRIBUTE_PROPERTIES = Ref{HDF5Properties}()
 
 const DEFAULT_PROPERTIES = HDF5Properties(H5P_DEFAULT, false, H5P_DEFAULT)
 
+const HAS_PARALLEL = Ref(false)
```

Comment:
Does this need to be a  constant ref?

Reply:
That value gets modified to `Ref(true)` if the `mpio.jl` file gets loaded via Requires, and if the HDF5 libraries include parallel support.

https://github.com/JuliaIO/HDF5.jl/blob/a12b66f9af28a76cbf7b099ed81709187744767f/src/mpio.jl#L4-L7

I thought it was a simple way to allow for this kind of dual behaviour (depending on whether code is loaded or not), but there may be a better way to do this.